src/options.c

```diff
@@ -247,38 +247,18 @@ static int G_GNUC_WGET_NORETURN print_help(G_GNUC_WGET_UNUSED option_t opt, G_GN
 	exit(0);
 }
 
-static char * wget_strnglob(const char *str, size_t n, int flags)
-{
-	static char *globstr;
-	static glob_t pglob;
-	static bool firstrun = true;
-	char *expanded_str = NULL;
-
-	if (!firstrun)
-		flags |= GLOB_APPEND;
-
-	globstr = wget_strmemdup(str, n);
-
-	if (glob(globstr, flags, NULL, &pglob) == 0) {
-		if (pglob.gl_pathc > 0) {
-			expanded_str = wget_strdup(pglob.gl_pathv[0]);
-		}
-		firstrun = false;
-	}
-	xfree(globstr);
-	return expanded_str;
-}
-
 static char *_shell_expand(const char *str)
 {
 	char *expanded_str = NULL;
 
 	if (*str == '~') {
 		char *pathptr = strchr(str, '/');
-		char *homestr = wget_strnglob(str, pathptr - str, GLOB_TILDE|GLOB_ONLYDIR|GLOB_MARK);
-		expanded_str = wget_str_asprintf("%s%s", homestr, pathptr+1);
-		xfree(homestr);
-	} else {
+		expanded_str = wget_strnglob(str, pathptr - str, GLOB_TILDE|GLOB_ONLYDIR|GLOB_NOCHECK);
```

Comment:
Maybe you can squash it right now, makes review easier on Github.
I'll have a closer look at it tomorrow...